ui/packages/catalog-ui-search/src/main/webapp/react-component/filter/filter.js

```diff
@@ -57,177 +55,109 @@ const FilterRemove = styled(Button)`
+    ) {
+      attribute = props.includedAttributes[0]
     }
 
-    componentDidMount = () => {
-      this.updateComparator()
-      this.updateSuggestions()
+    this.state = {
+      comparator,
+      attribute,
+      suggestions: props.suggestions || [],
+      value: (props.value && props.value[0]) || '',
+      isValid: props.isValid,
     }
+    props.onChange(this.state)
+  }
 
-    componentWillReceiveProps = ({ editing }) => {
-      this.setState({ editing })
-    }
-
-    render() {
-      const type = metacardDefinitions.metacardTypes[this.state.attribute].type
-      return (
-        <React.Fragment>
-          <FilterRearrange className="filter-rearrange">
-            <span className="cf cf-sort-grabber" />
-          </FilterRearrange>
-          <FilterRemove
-            buttonType={buttonTypeEnum.negative}
-            editing={this.state.editing}
-            onClick={this.props.onRemove}
-            icon="fa fa-minus"
-          />
-          <FilterAttribute
-            attribute={this.state.attribute}
-            includedAttributes={this.props.includedAttributes}
-            editing={this.state.editing}
-            onChange={this.updateAttribute}
-          />
-          <FilterComparator
-            comparator={this.state.comparator}
-            editing={this.state.editing}
-            type={type}
-            attribute={this.state.attribute}
-            onChange={comparator => {
-              this.setState({ comparator })
-              this.props.model.set('comparator', comparator)
-            }}
-          />
-
-          <FilterInput
-            suggestions={this.state.suggestions}
-            attribute={this.state.attribute}
-            comparator={this.state.comparator}
-            editing={this.state.editing}
-            model={this.props.model}
-            type={type}
-            isForm={this.props.isForm}
-            isFormBuilder={this.props.isFormBuilder}
-          />
-          <ExtensionPoints.filterActions
-            model={this.props.model}
-            metacardDefinitions={metacardDefinitions}
-            options={this.props}
-          />
-        </React.Fragment>
-      )
-    }
+  componentDidMount() {
+    this.updateSuggestions()
+  }
 
-    updateComparator = (attribute = this.state.attribute) => {
-      const currentComparator = this.props.model.get('comparator')
-      const value = Common.duplicate(this.props.model.get('value'))
-      const propertyJSON = generatePropertyJSON(
-        value,
-        attribute,
-        currentComparator
-      )
+  render() {
+    return (
+      <Root>
+        <FilterRearrange className="filter-rearrange">
+          <span className="cf cf-sort-grabber" />
+        </FilterRearrange>
+        <FilterRemove
+          buttonType={buttonTypeEnum.negative}
+          editing={this.props.editing}
+          onClick={this.props.onRemove}
+          icon="fa fa-minus"
+        />
+        <FilterAttribute
```

Comment:
✏️ for filter attribute, comparator, and input, you can pass through your props and state.
```
const props = {...this.state, ...this.props}
<Component {...props}>
```

Reply:
Probably won't be relevant after it's converted to a functional component, but I'm curious of this anyways

:question: How does react handle unused props?  For example if I make this change to FilterAttribute:
If value or comparator changes, will react re-render FilterAttribute because those props changed, or is it smart enough to know that FilterAttribute doesn't actually care about those props?

Reply:
I'm not sure. I'm guessing it would go through to check which parts of the dom needs rendering, but wouldn't actually do any new dom manipulation 
https://stackoverflow.com/a/46737063

Reply:
Ah cool, so it looks like functional components re-render every time regardless if you don't use React.memo.